src/Tools/Common/Commands/Utils.cs

```diff
@@ -59,66 +69,66 @@ public static bool ValidateArgumentsForChildProcess(int processId, string name,
-                Console.WriteLine("Only one of the --name, --process-id, or --diagnostic-port options may be specified.");
-                return false;
-            }
-            else if (processId != 0 && !string.IsNullOrEmpty(name))
-            {
-                Console.WriteLine("Only one of the --name or --process-id options may be specified.");
-                return false;
-            }
-            else if (processId != 0 && !string.IsNullOrEmpty(port))
-            {
-                Console.WriteLine("Only one of the --process-id or --diagnostic-port options may be specified.");
-                return false;
-            }
-            else if (!string.IsNullOrEmpty(name) && !string.IsNullOrEmpty(port))
+            else if ( processId != 0 && (!string.IsNullOrEmpty(name) || !string.IsNullOrEmpty(port) || !string.IsNullOrEmpty(dsrouter))
+                    || !string.IsNullOrEmpty(name) && (!string.IsNullOrEmpty(port) || !string.IsNullOrEmpty(dsrouter))
+                    || !string.IsNullOrEmpty(port) && !string.IsNullOrEmpty(dsrouter))
             {
-                Console.WriteLine("Only one of the --name or --diagnostic-port options may be specified.");
+                Console.WriteLine("Only one of the --name, --process-id, --diagnostic-port, or --dsrouter options may be specified.");
                 return false;
             }
-            // If we got this far it means only one of --name/--diagnostic-port/--process-id was specified
+            // If we got this far it means only one of --name/--diagnostic-port/--process-id/--dsrouter was specified
             else if (!string.IsNullOrEmpty(port))
             {
                 return true;
             }
             // Resolve name option
             else if (!string.IsNullOrEmpty(name))
             {
-                processId = CommandUtils.FindProcessIdWithName(name);
-                if (processId < 0)
+                if ((processId = FindProcessIdWithName(name)) < 0)
                 {
                     return false;
                 }
             }
-            else if (processId == 0)
+            else if (!string.IsNullOrEmpty(dsrouter))
             {
-                Console.WriteLine("One of the --name, --process-id, or --diagnostic-port options must be specified when attaching to a process.");
-                return false;
+                ConsoleColor currentColor = Console.ForegroundColor;
+                Console.ForegroundColor = ConsoleColor.Yellow;
+                Console.WriteLine("WARNING: dotnet-dsrouter is a development tool not intended for production environments." + Environment.NewLine);
+                Console.ForegroundColor = currentColor;
+
+                if (dsrouter != "ios" && dsrouter != "android" && dsrouter != "ios-sim" && dsrouter != "android-emu")
+                {
+                    Console.WriteLine("Invalid value for --dsrouter. Valid values are 'ios', 'ios-sim', 'android' and 'android-emu'.");
```

Comment:
maybe --dsrouter should accept the full command line that you can pass to it, not limiting it to the defaults? That way you could run parallel dsrouter instances at the same time using different set of ports.

Reply:
Although I'm not opposed to that, do you expect it will feel simple enough that customers would want to use it and you can easily document it? I assume you'd need to use some quotation syntax to nest a 2nd command-line inside the first one and different shells have different quotation rules. At that point it might be easier to just run dsrouter explicitly as a 2nd command.

Reply:
Yeah, maybe we should start simple and see how far it takes us.